### PR TITLE
Improve vectortranslate options

### PR DESCRIPTION
**Pull request by agila5**
Just the first prototype. Will add docs, tests, examples and the list approach suggested by @Robinlovelace. Example: 

``` r
# packages
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Check the package website, https://docs.ropensci.org/osmextract/, for more details.
library(sf)
#> Linking to GEOS 3.9.0, GDAL 3.2.1, PROJ 7.2.1

# create buffer
la_valletta = st_buffer(st_sfc(st_point(c(456113.1, 3972853)), crs = 32633), 5000)

# import
test <- oe_get(
  "Malta", 
  vectortranslate_options = c(
    "-where", "highway IN ('primary', 'secondary', 'tertiary', 'unclassified')",
    "-t_srs", "EPSG:32633", 
    "-clipdst", st_as_text(la_valletta), 
    "-nlt", "PROMOTE_TO_MULTI"
  )
)
#> The input place was matched with: Malta
#> The chosen file was already detected in the download directory. Skip downloading.
#> Start with the vectortranslate operations on the input file!
#> 0...10...20...30...40...50...60...70...80...90...100 - done.
#> Finished the vectortranslate operations on the input file!
#> Reading layer `lines' from data source `C:\Users\Utente\Documents\osm-data\geofabrik_malta-latest.gpkg' using driver `GPKG'
#> Simple feature collection with 1206 features and 9 fields
#> Geometry type: MULTILINESTRING
#> Dimension:     XY
#> Bounding box:  xmin: 451113.3 ymin: 3967859 xmax: 460364.8 ymax: 3976642
#> Projected CRS: WGS 84 / UTM zone 33N

plot(st_geometry(test))
```

<sup>Created on 2021-05-06 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

@Robinlovelace to be honest, the vectortranslate options are not that different from the example https://github.com/ropensci/osmextract/issues/178#issue-813430637, but I don't see any obvious method to specify the options using a much less verbose (and simple) way.

### CODE REVIEW

**Comment by Robinlovelace** [comment]
Looks like a good start. In terms of verbosity and simplicity, I think this could be neat and a few less things to remember - only marginal though:

```
test <- oe_get(
  "Malta", 
  vectortranslate_options = list(
    where = "highway IN ('primary', 'secondary', 'tertiary', 'unclassified')",
    t_srs = "EPSG:32633", 
    clipdst = st_as_text(la_valletta), 
    nlt = "PROMOTE_TO_MULTI"
  )
)
```

Idea: one way would help users remember all the options could be to provide an object that is packaged in `data/vtranslist.rda` or similar that has something like this

```
list(
  clipdist = NULL,
  t_srs = NULL
)
```

Making it easier to produce inputs to vectortranslate without having to do a deep dive into GDAL docs. Overall, the approach looks good to me though. For modularity I think a new function could help but +s/-s both ways.

**Comment by agila5** [comment]
I plan to start working on this in the afternoon but I think that, for the 0.3.0 version, I will include only the character vector option (and ignore the list option)

> Idea: one way would help users remember all the options could be to provide an object that is packaged in data/vtranslist.rda or similar that has something like this

Can you add more details?

**Comment by agila5** [comment]
Hi @Robinlovelace! New approach that I think could close #178: 

``` r
# packages
library(sf)
#> Linking to GEOS 3.9.0, GDAL 3.2.1, PROJ 7.2.1
library(osmextract)
#> Data (c) OpenStreetMap contributors, ODbL 1.0. https://www.openstreetmap.org/copyright.
#> Check the package website, https://docs.ropensci.org/osmextract/, for more details.

# clipdst
la_valletta = st_sfc(st_point(c(456113.1, 3972853)), crs = 32633) %>%
  st_buffer(5000)

oe_get(
  "Malta", 
  vectortranslate_options = c(
    "-clipdst", st_as_text(la_valletta), 
    "-t_srs", "EPSG:32633", 
    "-nlt", "PROMOTE_TO_MULTI"
  )
)
#> The input place was matched with: Malta
#> The chosen file was already detected in the download directory. Skip downloading.
#> Start with the vectortranslate operations on the input file!
#> 0...10...20...30...40...50...60...70...80...90...100 - done.
#> Finished the vectortranslate operations on the input file!
#> Reading layer `lines' from data source 
#>   `C:\Users\Utente\Documents\osm-data\geofabrik_malta-latest.gpkg' 
#>   using driver `GPKG'
#> Simple feature collection with 12426 features and 9 fields
#> Geometry type: MULTILINESTRING
#> Dimension:     XY
#> Bounding box:  xmin: 451113.3 ymin: 3967853 xmax: 460689.7 ymax: 3977228
#> Projected CRS: WGS 84 / UTM zone 33N
```

<sup>Created on 2021-05-26 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

I like the "list" approach but I would add that for 0.4.0 version.

File: tests/testthat/test-vectortranslate.R [hunk 1]
```diff
@@ -52,11 +63,8 @@ test_that("vectortranslate is not skipped if force_download is TRUE", {
     its_pbf,
     download_directory = tempdir(),
     vectortranslate_options = c(
-      "-f", "GPKG",
-      "-overwrite",
-      "-lco", "GEOMETRY_NAME=geometry",
-      "-where", "highway IN ('service')",
-      "lines"
+      # the other options should be filled automatically
```
Comment (Robinlovelace) [inline tests/testthat/test-vectortranslate.R:66]: +1 to them being autocompleted.

**Review by Robinlovelace — APPROVED** [review]
Looks good. I think there could be a new argument like `boundary` that further simplifies the code, so it's something like

```
oe_get("ilse of wight", boundary = shp)
```

but this is definite progress so thumbs up from me. Great work!

**Comment by agila5** [comment]
> Looks good. I think there could be a new argument like boundary that further simplifies the code, so it's something like

The only problem I see here is that there are several options in `ogr2ogr` that could be interpreted as "spatial boundary" (i.e. `-spat`, `-clipsrc`, `-clipdst` , `-sql` + spatiallite) and I'm not sure what's the best option (probably `-spat`). Feel free to create a new issue if you want.

**Comment by Robinlovelace** [comment]
> The only problem I see here is that there are several options in `ogr2ogr` that could be interpreted as "spatial boundary" (i.e. `-spat`, `-clipsrc`, `-clipdst` , `-sql` + spatiallite) and I'm not sure what's the best option (probably `-spat`). Feel free to create a new issue if you want.

Good point. There could be a generic `clip = `sf_object` arg and then another like `clip_type = "spat"` by default. Will open a new issue.

**Pull request merged**